pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -218,10 +218,6 @@ func (c *Controller) NetworkGateways() []model.NetworkGateway {
 	c.networkManager.RLock()
 	defer c.networkManager.RUnlock()
 
-	if len(c.networkGatewaysBySvc) == 0 {
-		return nil
-	}
```

Comment:
this now returns an empty set instead of nil like it did before

perhaps we'd rather check length of the new set before returning, and return nil if that length is zero?

Reply:
unless `model.NetworkGatewaySet` defaults to nil? and `Merge` can take a nil and initiate? i have not checked out the code..

Reply:
Merge can take a nil, it just will be a no-op, similar to an empty set.
https://github.com/istio/istio/blob/master/pkg/util/sets/set.go#L72